setup.py

```diff
@@ -5,26 +5,42 @@
     long_description = fh.read()
 
 dask_deps = ["dask>=2.22.0,<2022.2.0", "distributed>=2.22.0,<2022.2.0"]
-# TODO: remove redis dependency when resolving GH#4398
-# Pin ray to < 1.13.0 to work around GH#4564
-# TODO(https://github.com/modin-project/modin/issues/4564): let ray go past 1.13.0.
-# Following https://github.com/ray-project/ray/pull/25648, pin protobuf < 4,
-# because versions >= 4.0.0 are incompatible with ray<1.13.0.
 ray_deps = [
-    "ray[default]>=1.4.0,<1.13.0",
+    "ray[default]>=1.4.0",
     "pyarrow>=4.0.1",
     "redis>=3.5.0,<4.0.0",
-    "protobuf<4.0.0",
 ]
 remote_deps = ["rpyc==4.1.5", "cloudpickle", "boto3"]
 spreadsheet_deps = ["modin-spreadsheet>=0.1.0"]
 sql_deps = ["dfsql>=0.4.2", "pyparsing<=2.4.7"]
 all_deps = dask_deps + ray_deps + remote_deps + spreadsheet_deps
 
+# Distribute 'modin-autoimport-pandas.pth' along with binary and source distributions.
+# This file provides the "import pandas before Ray init" feature if specific
+# environment variable is set (see https://github.com/modin-project/modin/issues/4564).
+cmdclass = versioneer.get_cmdclass()
+
+
+class AddPthFileBuild(cmdclass["build_py"]):
+    def _get_data_files(self):
+        return (super()._get_data_files() or []) + [
+            (".", ".", self.build_lib, ["modin-autoimport-pandas.pth"])
```

Comment:
Should we declare a var for modin-autoimport-pandas.pth as we use it in both AddPthFileBuild and AddPthFileSDist?

Reply:
fixed